### DISCUSSION TITLE
Permission denied (publickey).

### DISCUSSION THREAD

**Discussion by lom4f1** [sshnet/SSH.NET#1489] [Q&A]
In the past I used the method _Renci.SshNet.PasswordConnectionInfo_ for internal test of ssh connection. Everything worked fine.
Passing to _Renci.SshNet.PrivateKeyConnectionInfo_ I am not able to connect to remote host at all, cause of the exception:
`Exception trying to connect ssh host (userName@hostName): Renci.SshNet.Common.SshAuthenticationException: Permission denied (publickey).
 in Renci.SshNet.ClientAuthentication.Authenticate(IConnectionInfoInternal connectionInfo, ISession session)
 in Renci.SshNet.ConnectionInfo.Authenticate(ISession session, IServiceFactory serviceFactory)
 in Renci.SshNet.Session.Connect()
 in Renci.SshNet.BaseClient.CreateAndConnectSession()
 in Renci.SshNet.BaseClient.Connect()`
 
Using the shell everything works and the key seems to have no problem, but using the library I am not able to connect to the remote host.
I have also tried to use directly the pivate key content instead of referencing the file in the code, but nothing changes.
Any idea about how to make it works?

Thanks in advance.

**Comment by Rob-Hague** [comment]
No obvious answer I'm afraid

Can you paste some sample code?
What command are you using in the shell where everything works?
What kind of key is it? (RSA, ECDSA, etc.)
What is the first line of the key file? e.g. ----- BEGIN XXX PRIVATE KEY -----

**Reply by lom4f1** [reply]
The code is something like this:
```c#
string keyFilePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile).ToString(),@".ssh\id_rsa");
var privateKey = new Renci.SshNet.PrivateKeyFile(keyFilePath);
var keyFiles = new[] { privateKey };
ClientSSH = new Renci.SshNet.SshClient(hostName,userName,keyFiles);
```
Manual command is:
`ssh -i ./id_rsa -v username@hostname`
The key starts with:
`-----BEGIN RSA PRIVATE KEY-----`

BTW, at the end, I asked for help to the server owner. He checked logs server side and the issue was related to server configuration. I do not know the reason why it could accept the key using ssh command line, but not the same key using the code.
Error sever side was something like:
`userauth_pubkey: key type ssh-rsa not in PubkeyAcceptedAlgorithms [preauth]`

So, at the end, generating a new keypair (this time ssh-ed25519), everything worked properly.
Thank you anyway for attention.

**Reply by Rob-Hague** [reply]
Are you on a version of the package older than 2023.0.0? That could explain it

**Reply by lom4f1** [reply]
No I updated the library to the 2024 version. But the behavior didn't change.

**Comment by darkoperator** [comment]
Also any logs from the server side?

**Reply by lom4f1** [reply]
Only this: `userauth_pubkey: key type ssh-rsa not in PubkeyAcceptedAlgorithms [preauth]`
